src/RFC-0205_LedgerTransactions.md

```diff
@@ -0,0 +1,148 @@
+
+All [UTXO]s created will be created with a script `PushPubkey(K_S)`. The key (\\( k_s \\) ) is created as follows:
+$$
+\begin{aligned}
+s = Hash(k_i * A) \\\\
+k_S = s + a \\\\
+K_S = s \cdot G + A
+\end{aligned}
+$$
+
+The blinding factor (\\( k_i \\) ) is used as a random nonce when creating the script key. This means the helper can create the public key without the signer present, and the signer can then at a later stage create the private key from the nonce. The key pair (\\( a, A \\) ) is the master private key from the signer. The private key (\\( a \\) ) is kept secret by the signer at all times.
+
+### Initialization
+Adding a hardware wallet to a wallet we need to ensure that all keys are only derived from a single seed phrase provided by the hardware wallet. 
+
+Helper asks signer for master helper key (\\( k_H \\) ). 
+This key is derived from the signer seed phrase.
+
+### Transaction receiving
+When a transaction is received the helper constructs the new [UTXO] with its Rangeproof. Choosing a new (\\( k_i \\) ) for the [UTXO], it calculates a new \\( s \\) and \\( K_S \\). It attaches the script `PushPubkey(K_S)` to output.
+
+### Transaction sending
+When the user wants to send a transaction, the helper retrieves the desired [UTXO]. The helper gives asks the signer to sign the transaction. 
+The singer calculates \\( s \\) from \\( a, K_i \\) and calculates \\( k_s \\) to sign the transaction. 
+The signer creates a random nonce \\( k_O \\) to use for the script_offset. It signs the metadata signature with \\( k_O \\), and supplies the script_offset to the helper. 
+The helper can attach the correct signatures to the [UTXO]s and ship the transaction.
+
+### Receiving normal one-sided transaction
+This can be done by the helper asking the signer for a public key. And advertising this public key as the destination public key for a 1-sided transaction. 
+The helper can scan the blockchain for this public key. 
+
+### Receiving one-sided-stealth
+`TODO`
+
+### Output recovery
+When creating outputs the wallet encrypts the blinding factor (\\k_i \\) and value \\( v \\) with a key derived from (\\( k_H \\) ).
+Because the key (\\( k_H \\) ) is calculated from the seed phrase of the signer, this will be the same each time. The helper can try to decrypt each scanned output, when it is successful it knows it has found its own output. 
+
+## Security
+Because the script key is required for spending, it is the only key that needs to be kept secret. 
+The following table explains what an attacker can do upon learning a key from the helper
+
+| key         | Worst case scenario          | 
+|:------------|:-----------------------------|
+| \\( k_H \\) | Can view all transactions and values made by the wallet |
+| \\( k_i \\) | Can try and brute force the value of the transaction |
```

Comment:
This can also occur if `K_i` were leaked.

Reply:
Yeah this table is here for that regardless of how the key is leaked or learned

Reply:
Sure, just noting that you don't need `k_i` to do the brute forcing of the value, as `K_i` is sufficient for it too.